commit b5ffa2c1a641a9fc244fff89ffd983b74432b272
Author: Ray Lau <ray.lau@lowrisc.org>
Date:   Tue May 19 16:12:02 2026 +0100

    [rtl] Fix ethernet Tx packets offset by 3 bytes
    
    tx_axis_tvalid is asserted for an extra 3 cycles after the last transfer,
    causing data in every packet sent after the 1st packet to be offset by
    3 bytes.
    
    The fix ensures the tvalid provided to rgmii_soc (tx_axis_tvalid_gated)
    is 0 when tx_enable_i is 0, which prevents the issue.

diff --git a/framing_top.sv b/framing_top.sv
index 7d1ca13..8723305 100644
--- a/framing_top.sv
+++ b/framing_top.sv
@@ -66,6 +66,7 @@ reg        byte_sync, sync, irq_en, tx_busy;
         */
         reg         tx_axis_tvalid;
         reg         tx_axis_tvalid_dly;
+        wire        tx_axis_tvalid_gated;
         reg 	    tx_axis_tlast;
         wire [7:0]  tx_axis_tdata;
         wire        tx_axis_tready;
@@ -288,6 +289,7 @@ always @(posedge clk_int)
 	      tx_axis_tvalid_dly <= 1'b0;
       end
  
+   assign tx_axis_tvalid_gated = tx_axis_tvalid & tx_enable_i;
    always @(posedge rx_clk)
      if (rst_int)
        begin
@@ -330,7 +332,7 @@ rgmii_soc rgmii_soc1
    .phy_pme_n(phy_pme_n),
    .mac_gmii_tx_en(mac_gmii_tx_en),
    .tx_axis_tdata(tx_axis_tdata),
-   .tx_axis_tvalid(tx_axis_tvalid),
+   .tx_axis_tvalid(tx_axis_tvalid_gated),
    .tx_axis_tready(tx_axis_tready),
    .tx_axis_tlast(tx_axis_tlast),
    .tx_axis_tuser(tx_axis_tuser),
